godot-codegen/src/generator/classes.rs

```diff
@@ -182,9 +182,10 @@ fn make_class(class: &Class, ctx: &mut Context, view: &ApiView) -> GeneratedClas
                 #internal_methods
                 #constants
             }
-            impl crate::obj::GodotClass for #class_name {
+            impl crate::obj::WithBase for #class_name {
                 type Base = #base_ty;
-
+            }
+            impl crate::obj::GodotClass for #class_name {
```

Comment:
Not a huge fan of this, in the future it means implementing one more trait for the user when using the builder API. But I see why it's needed...

Is there maybe a way to reconcile this with [`WithBaseField`](https://godot-rust.github.io/docs/gdext/master/godot/obj/trait.WithBaseField.html)?

Reply:
We can add an associated `Base` type to `WithBaseField` and require it to match `WithBase::Base`. Adding then a blanked implementation for `impl WithBase for T where T: WithBaseField` would have the effect that only one of the two traits has to be implemented. 

A downside of this is though that the compiler will now generate an error like this:
```
error[E0277]: the trait bound `gen::classes::viewport::re_export::Viewport: obj::traits::WithBaseField` is not satisfied
   --> godot-core/src/gen/classes/viewport.rs:947:68
    |
947 |     unsafe impl crate::obj::Inherits < crate::engine::Object > for Viewport {
    |                                                                    ^^^^^^^^ the trait `obj::traits::WithBaseField` is not implemented for `gen::classes::viewport::re_export::Viewport`, which is required by `gen::classes::viewport::re_export::Viewport: obj::traits::GodotClass`
``` 
If none of the two traits is implemented. 

This happens because the compiler tries to satisfy the `GodotClass: WithBase` requirement via the blanked implementation.